module/assets/helper.js

```diff
@@ -0,0 +1,1075 @@
+            rawEvent.context.env = constants.common.USER;
+            rawEvent.edata.state = constants.common.TRANSFER_STATE;
+            rawEvent.edata.type = constants.common.OWNERSHIP_TRANSFER_TYPE;
+            rawEvent.edata.props = [];
+            let userObject = {
+              id: reqData.toUserProfile.userId,
+              type: constants.common.USER,
+            };
+            rawEvent.actor = userObject;
+            rawEvent.object = userObject;
+            rawEvent.context.pdata.pid = `${process.env.ID}.${constants.common.OWNERSHIP_TRANSFER_MODULE}`;
+
+            let telemetryEvent = await gen.utils.generateTelemetryEvent(
+              rawEvent
+            );
+            telemetryEvent.lname = constants.common.TELEMTRY_EVENT_LOGGER;
+            telemetryEvent.level = constants.common.INFO_LEVEL;
+            /**
+             * @telemetryEvent
+             * {
+                  timestamp: 2024-05-06T06:15:54.742Z,
+                  msg: '{"eid":"AUDIT","ets":1714976154741,"ver":"3.0","mid":"bf25ba06-4d13-4161-83b2-aec50ceb1a56","actor":{"id":"9bb884fc-8a56-4727-9522-25a7d5b8ea28","type":"User"},"context":{"channel":"","pdata":{"id":"projectservice","ver":"8.0.0","pid":"projectservice.ownerTransfer"},"env":"User","cdata":[],"rollup":{}},"object":{"id":"9bb884fc-8a56-4727-9522-25a7d5b8ea28","type":"User"},"edata":{"state":"Transfer","type":"OwnerTransferStatus","props":[]}}',
+                  lname: 'TelemetryEventLogger',
+                  tname: '',
+                 level: 'INFO',
+                  HOSTNAME: '',
+  '               application.home': ''
+                } 
+               */
+            await kafkaProducersHelper.pushTelemetryEventToKafka(
+              telemetryEvent
+            );
+          }
+          return resolve({
+            success: true,
+          });
+        } else {
+          return resolve({
+            success: true,
+          });
+        }
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+
+  /**
+   * Transfers programs from one user to another.
+   * @method transferPlatformRoles
+   * @param {Object} fromUserData - The user data to transfer programs from.
+   * @param {Object} toUserData   - The user data to transfer programs to.
+   * @param {Object} [reqData={}] - requesbody data.
+   * @param {boolean} [checkAssetInformation=false] - check asset information.
+   * @returns {Promise<Object>}   -A promise that resolves with an object indicating the success of the program transfer.
+   * @throws {Error}              -If an error occurs during the transfer process.
+   */
+
+  static transferPlatformRoles(
+    fromUserData,
+    toUserData,
+    reqData = {},
+    checkAssetInformation = false
+  ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let updatePlatformRoleQueries = [];
+        fromUserData.platformRoles.forEach(async (role) => {
```

Comment:
@praveenKDass Why do we have to do this in loops? One loop should update from user and update to user with required information.